Sources/NostrSDK/EventKind.swift

```diff
@@ -30,8 +30,9 @@ public enum EventKind: RawRepresentable, CaseIterable, Codable, Equatable, Hasha
 
     /// This kind of event should have a recipient pubkey tag.
     ///
-    /// See [NIP-04 - Direct Messages](https://github.com/nostr-protocol/nips/blob/master/04.md)
-    case directMessage
+    /// See [NIP-04 - Encrypted Direct Message](https://github.com/nostr-protocol/nips/blob/master/04.md)
+    /// > Warning: Deprecated in favor of [NIP-17 - Private Direct Messages](https://github.com/nostr-protocol/nips/blob/master/04.md).
+    case nip04EncryptedDirectMessage
```

Comment:
I get that we need to distinguish between types of direct messaging, but--and this is just a suggestion, you have more context than I do--what if we called the nip04 ones "legacy"? So it would be obvious to the client that this is "the old way". We could also mark it `@deprecated`.

Reply:
I think that's a fine suggestion.

There's one edge case I can think of, though. If the new NIP-17 DM kind gets deprecated in the future (not that I'm anticipating it anytime soon, but anything is possible in the Nostr world), we would have another "legacy" DM kind, making the "legacy" naming convention overloaded.

However, I think optimizing for developer ergonomics might make more sense here. I'll make that change.

Reply:
Right, it's one possible future to have many DM kinds as the protocol progresses. In that case though, I would recommend that the current best practice one would always be named the most simply, like you said for developer ergonomics. And then all the legacy types would have uglier names with sufficient distinction, such as `NIP04LegacyDirectMessageEvent`.